src/pybind/mgr/dashboard/module.py

```diff
@@ -197,6 +199,33 @@ def _configure(self):
             if not os.path.isfile(pkey_fname):
                 raise ServerConfigException('private key %s does not exist' % pkey_fname)
 
+            # Do some validations to the private key and certificate:
+            # - Check the type and format
+            # - Check the certificate expiration date
+            # - Check the consistency of the private key
+            # - Check that the private key and certificate match up
+            try:
+                with open(cert_fname) as f:
+                    x509 = crypto.load_certificate(crypto.FILETYPE_PEM, f.read())
+                    if x509.has_expired():
+                        self.log.warning('Certificate has been expired')
```

Comment:
does it make sense to report the path of cert in the exception and/or in the log message? like, print out the `cert_fname` ?

the same applies to `pkey_fname`.

Reply:
also, to improve the readability, how about extracting this part into a static method?

Reply:
Done